flink-runtime/src/main/java/org/apache/flink/streaming/api/transformations/OneInputTransformation.java

```diff
@@ -195,6 +196,11 @@ public boolean isInternalSorterSupported() {
 
     @Override
     public void enableAsyncState() {
-        // nothing to do.
+        OneInputStreamOperator<IN, OUT> operator =
+                (OneInputStreamOperator<IN, OUT>)
+                        ((SimpleOperatorFactory<OUT>) operatorFactory).getOperator();
+        if (!(operator instanceof AsyncStateProcessingOperator)) {
+            super.enableAsyncState();
```

Comment:
I am curious shouldn't this be an error?
As is the caller amkes this call anf would assume that async is enabled, but it is not.